app/src/main/java/org/greenstand/android/TreeTracker/fragments/NewTreeFragment.kt

```diff
@@ -13,25 +13,30 @@ import androidx.core.app.ActivityCompat
 import androidx.lifecycle.Observer
 import androidx.lifecycle.lifecycleScope
 import androidx.navigation.fragment.findNavController
-import kotlinx.android.synthetic.main.activity_main.*
+import java.util.UUID
+import kotlinx.android.synthetic.main.activity_main.toolbarTitle
 import kotlinx.android.synthetic.main.fragment_new_tree.*
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.launch
 import org.greenstand.android.TreeTracker.R
 import org.greenstand.android.TreeTracker.managers.FeatureFlags
+import org.greenstand.android.TreeTracker.managers.LocationDataCapturer
 import org.greenstand.android.TreeTracker.utilities.CameraHelper
 import org.greenstand.android.TreeTracker.utilities.ImageUtils
 import org.greenstand.android.TreeTracker.utilities.ValueHelper
 import org.greenstand.android.TreeTracker.utilities.vibrate
 import org.greenstand.android.TreeTracker.view.CustomToast
 import org.greenstand.android.TreeTracker.viewmodels.NewTreeViewModel
+import org.koin.android.ext.android.inject
 import org.koin.android.viewmodel.ext.android.viewModel
 import timber.log.Timber
 
 class NewTreeFragment :
     androidx.fragment.app.Fragment(), ActivityCompat.OnRequestPermissionsResultCallback {
 
     private val vm: NewTreeViewModel by viewModel()
+    private val locationDataCapturer by inject<LocationDataCapturer>()
+    private var newTreeUuid: UUID? = null
```

Comment:
Can we move these two fields to the viewmodel?

When they are in the view like this it makes it harder to test later on. Ideally the view just renders stuff.

Reply:
`locationDataCapturer` is in itself a model. It makes sense to move these behind view model. But in the case of `mapsFragment` the context doesn't have a natural flow. I will make the change and you might see what I am talking about.